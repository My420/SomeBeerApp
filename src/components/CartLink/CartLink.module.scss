@import '../../sass/variables.scss';

.container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.amount {
  display: flex;
  justify-content: center;
  align-items: center;

  width: 18px;
  height: 18px;

  border-radius: 50%;
  background-color: $chip-color;

  font-size: 14px;
  line-height: 14px;
}
