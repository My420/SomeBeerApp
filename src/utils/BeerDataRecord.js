import { Record } from 'immutable';

const BeerDataRecord = Record(
  {
    id: null,
    name: null,
    tagline: null,
    first_brewed: null,
    description: null,
    image_url: null,
    abv: null,
    ibu: null,
    target_fg: null,
    target_og: null,
    ebc: null,
    srm: null,
    ph: null,
    attenuation_level: null,
    volume: {},
    boil_volume: {},
    method: {},
    ingredients: {},
    food_pairing: [],
    brewers_tips: null,
    contributed_by: null
  },
  'BeerDataRecord'
);

export default BeerDataRecord;
