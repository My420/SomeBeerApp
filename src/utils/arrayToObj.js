/* eslint no-param-reassign: "error" */

const arrayToObj = arr => {
  return arr.reduce((prevValue, element) => {
    prevValue[element.id] = element;
    return prevValue;
  }, {});
};

export default arrayToObj;
