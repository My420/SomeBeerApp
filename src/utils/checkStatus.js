const checkStatus = response => {
  const responseStatus = +response.status;
  if (responseStatus >= 200 && responseStatus < 300) {
    return response;
  }
  throw new Error(response.statusText);
};

export default checkStatus;
