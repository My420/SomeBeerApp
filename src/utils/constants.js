export const SHOP_NAME = 'Some Beer Shop';

export const LINK = {
  HOME: '/',
  CATALOG: '/catalog',
  ROULETTE: '/roulette',
  ABOUT: '/about',
  FAVORITE: '/favorite',
  CART: '/cart'
};

export const NAVIGATION_LINKS = [
  [LINK.HOME, 'Home'],
  [LINK.CATALOG, 'Catalog'],
  [LINK.ROULETTE, 'Roulette!'],
  [LINK.FAVORITE, 'Favorite'],
  [LINK.ABOUT, 'About'],
  [LINK.CART, 'Cart']
];

export const BROWSER_WIDTH_RATIO = {
  LPC: 4,
  MPC: 3,
  TABLET_9: 2,
  TABLET_7: 1,
  MOBILE: 0
};

export const BROWSER_MIN_WIDTH = {
  LPC: 1200,
  MPC: 993,
  TABLET_9: 768,
  TABLET_7: 480,
  MOBILE: 320
};

export const SOCIAL_LINK = {
  GITHUB: 'https://github.com/My420',
  EMAIL: 'mailto:klinovitsky.aleksey@gmail.com',
  SKYPE: 'skype:live:klinovitsky.aleksey?chat'
};

export const AUTHOR = 'Klinovitsky Aleksey';

export const POPULAR_BEER__GET_REQUEST = 'POPULAR_BEER__GET_REQUEST';
export const POPULAR_BEER__GET_SUCCESS = 'POPULAR_BEER__GET_SUCCESS';
export const POPULAR_BEER__GET_FAILURE = 'POPULAR_BEER__GET_FAILURE';

export const API_HOST = 'https://api.punkapi.com/v2/';

export const POPULAR_BEER_AMOUNT = 8;

export const LOAD_ERROR_MESSAGE =
  'Could not download data, please refresh the page';

export const ERROR_TITLE = 'Oops! Something went wrong.';
export const ERROR_ACTION = 'Please refresh page or try again later.';
