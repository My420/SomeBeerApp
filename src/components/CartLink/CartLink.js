import React from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';
import styles from './CartLink.module.scss';
import withCart from '../HOC/CartWrapper/CartWrapper';
import { CART } from '../../utils/constants';

export const CartLink = ({ cartData }) => {
  // eslint-disable-next-line no-console
  console.log('render ===== CartLink');

  const itemAmount = cartData.size;

  return (
    <span className={styles.container}>
      <span className={styles.amount}>{itemAmount}</span>
      <span className={styles.name}>{CART}</span>
    </span>
  );
};

CartLink.propTypes = {
  cartData: PropTypes.instanceOf(Map).isRequired
};

export default withCart(CartLink);
