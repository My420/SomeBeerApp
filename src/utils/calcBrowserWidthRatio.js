import { BROWSER_MIN_WIDTH, BROWSER_WIDTH_RATIO } from './constants';

const calcBrowserWidthRatio = width => {
  if (width > BROWSER_MIN_WIDTH.LPC) {
    return BROWSER_WIDTH_RATIO.LPC;
  }
  if (width > BROWSER_MIN_WIDTH.MPC) {
    return BROWSER_WIDTH_RATIO.MPC;
  }
  if (width > BROWSER_MIN_WIDTH.TABLET) {
    return BROWSER_WIDTH_RATIO.TABLET_9;
  }
  if (width > BROWSER_MIN_WIDTH.MOBILE) {
    return BROWSER_WIDTH_RATIO.TABLET_7;
  }
  return BROWSER_WIDTH_RATIO.MOBILE;
};

export default calcBrowserWidthRatio;
