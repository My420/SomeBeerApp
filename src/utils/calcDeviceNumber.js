import { DEVICE_MIN_WIDTH, DEVICE_NUMBER } from './constants';

const calcDeviceNumber = width => {
  if (width > DEVICE_MIN_WIDTH.LPC) {
    return DEVICE_NUMBER.LPC;
  }
  if (width > DEVICE_MIN_WIDTH.MPC) {
    return DEVICE_NUMBER.MPC;
  }
  if (width > DEVICE_MIN_WIDTH.TABLET) {
    return DEVICE_NUMBER.TABLET;
  }
  return DEVICE_NUMBER.MOBILE;
};

export default calcDeviceNumber;
