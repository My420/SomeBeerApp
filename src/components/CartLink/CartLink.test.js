import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';
import { CartLink } from './CartLink';
import { CART } from '../../utils/constants';

describe('Test <CartLink /> component', () => {
  const cart = Map({
    a: 1,
    b: 2
  });

  const wrapper = shallow(<CartLink cartData={cart} />);

  test('should correct render <CartLink /> component ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render tag <span>  with className = container', () => {
    expect(wrapper.find('span.container')).toHaveLength(1);
  });

  test('should render tag <span>  with className = amount, inside container', () => {
    expect(wrapper.find('span.container').find('span.amount')).toHaveLength(1);
  });

  test('should render cart items amount', () => {
    expect(wrapper.find('span.amount').text()).toBe(`${cart.size}`);
  });

  test('should render tag <span>  with className = name, inside container', () => {
    expect(wrapper.find('span.container').find('span.name')).toHaveLength(1);
  });

  test('should render correct cart link text', () => {
    expect(wrapper.find('span.name').text()).toBe(CART);
  });
});
