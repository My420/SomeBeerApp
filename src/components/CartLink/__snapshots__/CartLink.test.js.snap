// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CartLink /> component should correct render <CartLink /> component  1`] = `
<span
  className="container"
>
  <span
    className="amount"
  >
    2
  </span>
  <span
    className="name"
  >
    Cart
  </span>
</span>
`;
